src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2150,6 +2167,87 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
+    //   the full path of the file to our terminal connection. Like conhost, if
+    //   the path contains a space, we'll wrap the path in quotes.
+    // Arguments:
+    // - e: The DragEventArgs from the Drop event
+    // Return Value:
+    // - <none>
+    winrt::fire_and_forget TermControl::_DoDragDrop(DragEventArgs const e)
+    {
+        if (e.DataView().Contains(StandardDataFormats::StorageItems()))
+        {
+            auto items = co_await e.DataView().GetStorageItemsAsync();
+            if (items.Size() > 0)
+            {
+                for (auto item : items)
+                {
+                    const auto fullPath = item.Path();
+                    const auto containsSpaces = std::find(fullPath.begin(),
+                                                          fullPath.end(),
+                                                          L' ') != fullPath.end();
+
+                    auto lock = _terminal->LockForWriting();
+                    if (containsSpaces)
+                    {
+                        WriteInput(L"\"");
+                    }
+                    WriteInput(fullPath);
+                    if (containsSpaces)
+                    {
+                        WriteInput(L"\"");
+                    }
+                }
+            }
+        }
+    }
+
+    // Method Description:
+    // - Synchronous handler for the "Drop" event. We'll dispatch the async
+    //   _DoDragDrop method to handle this, because getting information about
+    //   the file that was potentially dropped onto us muts be done off the UI
+    //   thread.
+    // Arguments:
+    // - e: The DragEventArgs from the Drop event
+    // Return Value:
+    // - <none>
+    void TermControl::_DragDropHandler(Windows::Foundation::IInspectable const& /*sender*/,
+                                       DragEventArgs const& e)
+    {
+        // Dispatch an async method to handle the drop event.
+        _DoDragDrop(e);
+    }
+
+    // Method Description:
+    // - Handle the DragOver event. We'll signal that the drag operation we
+    //   support is the "copy" operation, and we'll also customize the
+    //   appearance of the drag-drop UI, by removing the preview and setting a
+    //   custom caption. For more information, see
+    //   https://docs.microsoft.com/en-us/windows/uwp/design/input/drag-and-drop#customize-the-ui
+    // Arguments:
+    // Arguments:
+    // - e: The DragEventArgs from the DragOver event
+    // Return Value:
+    // - <none>
+    void TermControl::_DragOverHandler(Windows::Foundation::IInspectable const& /*sender*/,
+                                       DragEventArgs const& e)
+    {
+        // Make sure to set the AcceptedOperation, so that we can later recieve the path in the Drop event
+        e.AcceptedOperation(winrt::Windows::ApplicationModel::DataTransfer::DataPackageOperation::Copy);
+
+        // Sets custom UI text
+        e.DragUIOverride().Caption(RS_(L"DragFileCaption"));

```

Comment:
So this appears when you're dragging a file over the TermControl, and you just hover (don't let go of the file)?

Reply:
Yep, that's correct